// @sap/ux-specification/dist/src/specification/v4/webapp/manifest/sapUi5.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FIORI_FCL_ROUTER_CLASS = 'sap.f.routing.Router';
var FlexibleColumnLayoutAggregations;
(function (FlexibleColumnLayoutAggregations) {
    FlexibleColumnLayoutAggregations["BeginColumnPages"] = "beginColumnPages";
    FlexibleColumnLayoutAggregations["MidColumnPages"] = "midColumnPages";
    FlexibleColumnLayoutAggregations["EndColumnPages"] = "endColumnPages";
})(FlexibleColumnLayoutAggregations = exports.FlexibleColumnLayoutAggregations || (exports.FlexibleColumnLayoutAggregations = {}));
var FlexibleColumnLayoutType;
(function (FlexibleColumnLayoutType) {
    FlexibleColumnLayoutType["OneColumn"] = "OneColumn";
    FlexibleColumnLayoutType["TwoColumnsBeginExpanded"] = "TwoColumnsBeginExpanded";
    FlexibleColumnLayoutType["TwoColumnsMidExpanded"] = "TwoColumnsMidExpanded";
    FlexibleColumnLayoutType["MidColumnFullScreen"] = "MidColumnFullScreen";
    FlexibleColumnLayoutType["ThreeColumnsMidExpanded"] = "ThreeColumnsMidExpanded";
    FlexibleColumnLayoutType["ThreeColumnsEndExpanded"] = "ThreeColumnsEndExpanded";
    FlexibleColumnLayoutType["ThreeColumnsMidExpandedEndHidden"] = "ThreeColumnsMidExpandedEndHidden";
    FlexibleColumnLayoutType["ThreeColumnsBeginExpandedEndHidden"] = "ThreeColumnsBeginExpandedEndHidden";
    FlexibleColumnLayoutType["EndColumnFullScreen"] = "EndColumnFullScreen";
})(FlexibleColumnLayoutType = exports.FlexibleColumnLayoutType || (exports.FlexibleColumnLayoutType = {}));
